provisioning/miniprov/hare_mp/main.py

```diff
@@ -449,6 +449,10 @@ def init(args):
     try:
         url = args.config[0]
         utils = Utils(ConfStoreProvider(url))
+        provider = ConfStoreProvider(url)
+        machine_id = provider.get_machine_id()
+        if provider.get(f'node>{machine_id}>type') == 'server_node':
+           exit(0)
```

Comment:
Let's stop using `exit()` in functions outside of `main()`. One day we'll want to cover this code with unit tests and those voluntary `exit()` invocations will be a problem.
Please consider either just returning or raising an exception (which will turn into `sys.exit(error_code)` in `main` function).